### PR TITLE
fix energy requirement

### PR DESCRIPTION
**Pull request by jackbayliss**
The Energy requirement for research now checks against energy production now, rather than the current balance.

Which, hopefully should resolve https://github.com/lanedirt/OGameX/issues/208 https://github.com/lanedirt/OGameX/issues/174

I don't / haven't ever played this game, so hopefully I've got the logic / fix right. 

If not let me know 👍🏻

### CODE REVIEW

File: app/Services/PlanetService.php [hunk 1]
```diff
@@ -473,6 +473,9 @@ public function hasResources(Resources $resources): bool
         if (!empty($resources->deuterium->get()) && $this->deuterium()->get() < $resources->deuterium->get()) {
             return false;
         }
+        if (!empty($resources->energy->get()) && $this->energyProduction()->getFormattedLong() < $resources->energy->get()) {
```
Comment (lanedirt) [inline app/Services/PlanetService.php]: I'm not sure if this is a typo but I think the `getFormattedLong()` here should be `get()` instead.

Currently the formatted string (e.g. "5.33M") is being compared with the raw number (e.g. 5330000).

Other than this the change itself looks good to me. 👍

**Review by lanedirt — CHANGES_REQUESTED** [review]
Hi @jackbayliss,

Thanks for the PR! I have one small comment, see below. Could you take a look at this? Thanks again!

File: resources/views/ingame/ajax/object.blade.php [hunk 2]
```diff
@@ -96,7 +96,7 @@
                     @endif
                     @if (!empty($price->energy->get()))
                             <li class="resource energy icon sufficient tooltip js_hideTipOnMobile
-                        @if ($planet->energy()->get() < $price->energy->get())
+                        @if ($planet->energyProduction()->getFormattedLong() < $price->energy->get())
```
Comment (lanedirt) [inline resources/views/ingame/ajax/object.blade.php]: The comment above applies here as well.

File: app/Services/PlanetService.php (hunk 1 above)
Comment (jackbayliss) [reply app/Services/PlanetService.php]: Thanks! Will adjust, I copied the logic that was showing in the pop up. Leave it with me 👍

**Comment by lanedirt** [comment]
I just tested it on my local environment and looks like its working properly now for both mentioned tickets. Awesome, thanks a lot! 🚀  

Merging!

**Pull request merged**